.github/workflows/doxygen.yml

```diff
@@ -0,0 +1,104 @@
+name: Doxygen
+
+on:
+  push:
+    branches:
+      - master
+      - branch-DDS-3.*
+      - gh_wf_*
+    paths:
+      - '**'
+      # Don't run this workflow if the only files that changed are the
+      # following. Make sure this is the same as the "pull_request.paths" list.
+      - '!AUTHORS'
+      - '!PROBLEM-REPORT-FORM'
+      - '!LICENSE'
+      - '!Dockerfile**'
+      - '!.dockerignore'
+      - '!.mailmap'
+      - '!.github/workflows/lint.yml'
+      - '!etc/**'
+      - '!docs/**'
+      - '!hooks/**'
+      - '!tools/scripts/**'
+      - '!**.md'
+      - '!**.rst'
+      - '!**/.gitignore'
+      - '!**/.lint_config'
+      - '!**/README*'
+  pull_request:
```

Comment:
Do want to build Doxygen for pull requests? I could see it being handy to check for errors if we ever get the Doxygen build clean, but we're pretty far from that right now I think...